File: tests/fixtures/csharp-to-cpp-port/sample-app/Util/StringHelpers.cs
using System.Collections.Generic;
using System.Text;

namespace SampleApp.Util
{
    public static class StringHelpers
    {
        public static string PadId(int id)
        {
            return id.ToString().PadLeft(4, '0');
        }

        public static string JoinLines(IEnumerable<string> lines)
        {
            var sb = new StringBuilder();
            bool first = true;
            foreach (string line in lines)
            {
                if (!first) sb.Append('\n');
                sb.Append(line);
                first = false;
            }
            return sb.ToString();
        }

        public static bool IsBlank(string value)
        {
            return string.IsNullOrWhiteSpace(value);
        }
    }
}

File: tests/fixtures/csharp-to-cpp-port/expected-cpp/Util/StringHelpers.cpp
// Ported from Util/StringHelpers.cs
#include "Util/StringHelpers.h"

#include <cwctype>

#include "PortSupport.h"

namespace SampleApp::Util::StringHelpers {

std::wstring PadId(int id)
{
    std::wstring text = PortSupport::ToWString(id);
    if (text.size() < 4) {
        text.insert(0, 4 - text.size(), L'0');
    }
    return text;
}

std::wstring JoinLines(const std::vector<std::wstring>& lines)
{
    std::wstring result;
    bool first = true;
    for (const std::wstring& line : lines) {
        if (!first) result.push_back(L'\n');
        result += line;
        first = false;
    }
    return result;
}

bool IsBlank(const std::wstring& value)
{
    for (wchar_t ch : value) {
        if (!std::iswspace(ch)) return false;
    }
    return true;
}

} // namespace SampleApp::Util::StringHelpers
